documentation/docs/extensibility.md

```diff
@@ -1,86 +1,214 @@
+    Therefore, it is not possible to use extensions on this stage.
+
+### Practical example
+
+For a more practical example, you can use extensions in the SAP Cloud SDK pipeline to add custom linters to your pipeline.
+
+A linter is a tool that can check the source code for certain stylistic criteria. Many teams choose to use a linter to ensure a common programming style.
+
+For example, if you want to use [Checkstyle](https://checkstyle.sourceforge.io/) in your codebase, you might use an extension similar to this one in a file called `.pipeline/extensions/lint.groovy` in your project:
+
+```groovy
+def call(Map parameters) {
+
+    parameters.originalStage() // Runs the built in linters
+
+    mavenExecute(
+        script: parameters.script,
+        flags: '--batch-mode',
+        pomPath: 'application/pom.xml',
+        m2Path: s4SdkGlobals.m2Directory,
+        goals: 'checkstyle:checkstyle',
+    )
+
+    recordIssues blameDisabled: true,
+        enabledForFailure: true,
+        aggregatingResults: false,
+        tool: checkStyle()
+}
+
+return this
+```
+
+This code snippet has three components, let's see what is happening here:
+
+Firstly, we run the original stage.
+This runs the SAP UI5 Best Practices linter as this is a standard feature of SAP Cloud SDK pipeline.
+
+Secondly, we run the checkstyle maven plugin using the `mavenExecute` Jenkins library step as provided by project "Piper".
+This serves as an example for how flexible you can re-use what project "Piper" already provides in your extension.
+
+Finally, we use the Jenkins [Warnings NG plugin](https://plugins.jenkins.io/warnings-ng/) and its step `recordIssues` to make the findings visible in the Jenkins user interface.
+
+This example can be adapted for other linters of your choice.
+Be sure to checkout the _Library steps_ section of this documentation if you want to do this.
+Project "Piper" provides some basic building blocks such as `dockerExecute` and the already mentioned `mavenExecute` which might be helpful.
+
+## 2. Modified Ready-Made Pipeline
+
+This option describes how you can copy and paste one of the centrally maintained pipelines to make changes not possible otherwise.
+For example, you can't change the order of stages and the stages that run in parallel or add new stages to a centrally maintained pipeline.
+This might be done for an individual project (in the `Jenkinsfile`), or in a separate Git repository so it can be used for multiple projects.
+
+### Single project
+
+The default `Jenkinsfile` of centrally maintained pipelines does nothing else but loading the pipeline and running it.
+This is convenient but limits the modifiable aspects of the pipeline.
+
+If one of your projects uses the pipeline, the easiest way to do this modification is to copy the pipeline into your `Jenkinsfile`.
+
+The basic structure of your `Jenkinsfile` should be the following:
+
+```groovy
+@Library(/* Shared library definition, see below */) _
+
+call script: this
```

Comment:
> I guess in the end we should make it as easy as possible for users.

Yes, this is what I'm aiming for. I would argue that copying the template and filling in the call method is easier compared to the search and replace you would otherwise need to do, but I'm open to be convinced otherwise.